### ISSUE TITLE
Dashboard A11y: Move focus on successful story deletion

### ISSUE THREAD

**Issue by BrittanyIRL** [GoogleForCreators/web-stories-wp#4198]
## Bug Description

When a story is deleted in the dashboard with a keyboard, focus moves to the top of the dashboard. 

## Expected Behaviour

The existing functionality is confusing, we should move the focus to the next story I think when a story is successfully deleted. If there's no next story to focus on then the previous one should be focused. 

## Steps to Reproduce

Use the keyboard to delete a story in the dashboard, see that focus is moved on successful delete.

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by carloskelly13** [comment]
PR https://app.zenhub.com/workspace/o/google/web-stories-wp/issues/4317 has the commits for this ticket.

**Comment by csossi** [comment]
Verified in QA
